### ISSUE TITLE
Railways have no labels

### ISSUE THREAD

**Issue by claysmalley** [osm-americana/openstreetmap-americana#863]
Railways are completely missing from the `transportation_name` layer in OpenMapTiles, which prevents OpenStreetMap-Americana from labeling them.

[![image](https://user-images.githubusercontent.com/1732117/235496436-2e90ff54-8293-453b-b7e3-fb356851801c.png)](https://americanamap.org/#map=15.94/41.152551/-83.409976)

Most other OSM renderers, like OSM-Carto, label railways with the content of `name=*`:

[![image](https://user-images.githubusercontent.com/1732117/235496407-1dcd622d-828d-4b00-a3f7-e5fc7145b4d9.png)](https://www.openstreetmap.org/#map=17/41.15238/-83.41001)

Most American maps, however, label railroads with their respective reporting marks, if there are any labels at all.

Ideally, OSM-Americana should have railroads labeled with both the reporting marks and the name of the line. Railroads often assign names to their lines that are already in use by another railroad company on a different, nearby line, which makes it necessary to disambiguate by operator.

I can think of two obvious options to concatenate name and operator: `ABCD Wherever Subdivision`, or `Wherever Subdivision (ABCD)`. I personally prefer the latter, as the parentheses make it possible to distinguish the reporting marks from the name when one of them is unknown or intentionally blank.

**Comment by 1ec5** [comment]
As far as I’m aware, the reporting marks for North American railroads are in `operator:short`, which is supported by NSI, though it’s being challenged in osmlab/name-suggestion-index#7954. If railroad tracks aren’t reliably tagged with `operator:short`, then we’d need to look up the reporting marks in Wikidata based on `operator:wikidata` (which would be more robust anyways).

**Comment by Hiausirg** [comment]
"If railroad tracks aren’t reliably tagged with operator:short, then we’d need to look up the reporting marks in Wikidata based on operator:wikidata (which would be more robust anyways)."
Why don't just look it up using the normal (&existing) operator tags? Wikidata can generate such a "operator"->"operator:short" table: https://w.wiki/6Rg8
Looks far more reliable & easier to me.

Besides: ...I don't really have a clue how programming works, but how is this "look up reporting marks using operator:wikidata" supposed to work? Should the Americana map contact wikidata everytime someone loads the map, or the map is generated, to get those operator codes (which never change anyway)? Having an internal table to look up such codes sounds far easier and especially faster than connecting with wikidata every time.

**Comment by 1ec5** [comment]
> Should the Americana map contact wikidata everytime someone loads the map, or the map is generated

Not necessarily. OpenMapTiles/Planetiler already download a database dump of Wikidata in order to map `wikidata` values to labels for use in names etc. when generating the vector tiles. As long as the vector tiles have this information in there, there’s no need for the client (Americana in this case) to fetch anything from Wikidata on demand. This is currently only implemented for names of features in various languages, but there are some other possibilities around `*:wikidata` as well. That’s all I’m referring to.

(Americana does currently query Wikidata for the names of highway route networks on demand when opening the Legend panel. This is currently optimized for facilitating more translations of these route network names, but it isn’t the only way to do it. The load would be quite manageable for Wikidata even if Americana were much more popular.)

Americana already has a very large lookup table for mapping highway route `network` values to route shield images, although that’s easily justifiable because the mapping is somewhat specific to Americana’s artistic style. For something not specific to a particular style, it would make more sense for any lookup table to live on the server side. But OpenMapTiles has set a very high bar for introducing new lookup tables, since it’s trying to be flexible with respect to a continually evolving OSM. I’d imagine that part of the discussion would be on the merits of looking up spelled-out `operator`s versus looking up `operator:wikidata` values using [this very similar query](https://w.wiki/6g66).

**Comment by 1ec5** [comment]
Anyways, I see enough maps that label railroad tracks by their passenger services that I’m keen to see names like “Amtrak” incorporated into the labeling somehow. This would require us to fetch `network:wikidata` from the public transportation route relations. And yes, it would be `network` rather than `network:wikidata`, because `network` is intentionally ambiguous.

**Comment by wmisener** [comment]
> Anyways, I see enough maps that label railroad tracks by their passenger services that I’m keen to see names like “Amtrak” incorporated into the labeling somehow.

Maybe related to the request to highlight intercity passenger rail lines: #745
